### ISSUE TITLE
test_unistd_unlink is failing on the emscripten-releases builder (and also for me locally).

### ISSUE THREAD

**Issue by sbc100** [emscripten-core/emscripten#15492]
Since #15455 landed I'm not able to successfully run this test, and nor is the emscripten-releases builder.

What I'm not clear about is how come this is not failing on the emscripten CI.  The failure looks like this:

```
FAIL: test_realpath_nodefs (test_other.other)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\b\s\w\ir\cipd_bin_packages\cpython3\bin\Lib\unittest\case.py", line 60, in testPartExecutor
    yield
  File "C:\b\s\w\ir\cipd_bin_packages\cpython3\bin\Lib\unittest\case.py", line 676, in run
    self._callTestMethod(testMethod)
  File "C:\b\s\w\ir\cipd_bin_packages\cpython3\bin\Lib\unittest\case.py", line 633, in _callTestMethod
    method()
  File "C:\b\s\w\ir\k\install\emscripten\tests\test_other.py", line 6270, in test_realpath_nodefs
    self.assertContained('Resolved: /working/TEST_NODEFS.txt', self.run_js('a.out.js'))
  File "C:\b\s\w\ir\k\install\emscripten\tests\common.py", line 698, in run_js
    self.fail('JS subprocess failed (%s): %s.  Output:\n%s' % (error.cmd, error.returncode, ret))
  File "C:\b\s\w\ir\cipd_bin_packages\cpython3\bin\Lib\unittest\case.py", line 753, in fail
    raise self.failureException(msg)
AssertionError: JS subprocess failed (C:\b\s\w\ir\cache\builder\emscripten-releases\node-v14.15.5-win-x64\node.exe --stack-trace-limit=50 --unhandled-rejections=throw --trace-uncaught C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js): 7.  Output:
Aborted(Assertion failed: undefined)
Aborted(RuntimeError: Aborted(Assertion failed: undefined))
C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:141
      throw ex;
      ^

RuntimeError: Aborted(RuntimeError: Aborted(Assertion failed: undefined))
    at abort (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1515:11)
    at ___syscall_readlink (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:4471:69)
    at <anonymous>:wasm-function[41]:0xf7a
    at <anonymous>:wasm-function[12]:0x774
    at main (<anonymous>:wasm-function[10]:0x3d7)
    at C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1559:22
    at callMain (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5099:15)
    at doRun (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5156:23)
    at run (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5171:5)
    at runCaller (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5077:19)
    at removeRunDependency (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1488:7)
    at receiveInstance (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1648:5)
    at receiveInstantiationResult (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1665:5)
Thrown at:
    at abort (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1515:11)
    at ___syscall_readlink (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:4471:69)
    at :1:3963
    at :1:1909
    at main (:1:984)
    at C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:1559:22
    at callMain (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5099:15)
    at doRun (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5156:23)
    at run (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5171:5)
    at runCaller (C:\b\s\w\ir\tmp\t\tmp3j_wc7cp\emscripten_test_other_86z_eylk\a.out.js:5077:19)

```

**Comment by sbc100** [comment]
Does `./tests/runner wasm0.test_unistd_unlink` work for you locally @ethanalee ?

**Comment by sbc100** [comment]
I think maybe we should temporarily revert  https://github.com/emscripten-core/emscripten/pull/15455

**Comment by sbc100** [comment]
Actually I found a fairly simple fix: #15494

**Comment by ethanalee** [comment]
> Does `./tests/runner wasm0.test_unistd_unlink` work for you locally @ethanalee ?

It ran locally for me. I think the issue may have been not cleaning up the file after the test.
